xrpl/core/binarycodec/types/amount.py

```
@@ -107,6 +109,44 @@ def verify_iou_value(issued_currency_value: str) -> None:
+        raise XRPLBinaryCodecException(f"{mpt_value} is an invalid MPT amount.")
+
+    decimal = None
+    try:
+        # Check if the value is hexadecimal
+        if mpt_value.startswith("0x") or _HEX_REGEX.fullmatch(mpt_value):
+            decimal = Decimal(int(mpt_value, 16))
+        else:
+            # Check if mpt_value can be converted to a Decimal and within valid range
+            decimal = Decimal(mpt_value)
+    except (InvalidOperation, ValueError):
+        raise XRPLBinaryCodecException(f"{mpt_value} is not a valid MPT amount.")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)